cozmo_soar.py

```diff
@@ -648,6 +654,153 @@ def on_input_phase(self, input_link: sml.Identifier):
+                t2 = +1.0 if t2 > +1.0 else t2
+                t2 = -1.0 if t2 < -1.0 else t2
+                Y = math.asin(t2)
+
+                t3 = +2.0 * (w * z + x * y)
+                t4 = +1.0 - 2.0 * (y * y + z * z)
+                Z = math.atan2(t3, t4)
+                return (X, Y, Z)
+
+            eul = quaternion_to_euler(quat)
+
+            if type(obj).__name__ == '_proxy_LightCube':
+                LIGHT_CUBE_LENGTH = 43
+                depth = LIGHT_CUBE_LENGTH
+                width = LIGHT_CUBE_LENGTH
+                height = LIGHT_CUBE_LENGTH
+            
+            else:
+                depth = obj.x_size_mm
+                width = obj.y_size_mm
+                height = obj.z_size_mm
+
+            # generate [GEOMETRY] for svs str by generating vertices from position and vectors of dimensions      
+            def combinations(possibilities):
+                """
+                Takes in array of the form [(tuple), (tuple), (tuple)] and returns all the possible combinations.  You can think of each tuple as possibilities of a value at a certain place (index in top level array).  This returns all possible combinations of those values in each place.
+                """
+                # will be modified during recursion
+                combs = []
+                def comb_helper(arr, i, baton):
+                    """
+                    Modifies combs variable above
+                    """
+
+                    for elt in arr[i]:
+                        baton.append(elt)
+                        if i < len(arr) - 1:
+                            comb_helper(arr, i + 1, baton)
+                        else:
+                            combs.append(copy.deepcopy(baton))
+                        baton.pop()
+                comb_helper(possibilities, 0, [])
+                return combs
+
+            # vectors from center of object to vertices, unrotated
+            vectors = np.transpose(np.array(combinations([(depth/2.0, -depth/2.0), (width/2.0, -width/2.0), (height/2.0, -height/2.0)])))
+            rotation_matrix = transforms3d.euler.euler2mat(eul[0], eul[1], eul[2], axes='sxyz')
+            rotated_vectors = np.transpose(np.matmul(rotation_matrix, vectors))
+            
+            # rotated vertices = position array + rotated vectors
+            vertices = np.transpose(np.array([[ai + bi for ai, bi in zip(pos_arr, vector)] for vector in rotated_vectors]))
+            
+            # generate [TRANSFORM] portion of svs string
+            scalar = 1000.0   # turns navmap oordinates in mm to SVS coordinates in m
+            gen_transform = lambda pos, rot: f"p {pos[0]/scalar} {pos[1]/scalar} {pos[2]/scalar} r {rot[0]} {rot[1]} {rot[2]}"
+            geo_str = "v"
+            for vertex in vertices:
+                geo_str += f' {vertex[0]/scalar} {vertex[1]/scalar} {vertex[2]/scalar}'
+            svs_str = f"{geo_str} {gen_transform(pos_arr, eul)}"
```

Comment:
Rather than calling the lambda, just use an extra couple lines to generate the strings you need.
```suggestion
            pos_transform_str = f"p {pos_arr[0]/scalar} {pos_arr[1]/scalar} {pos_arr[2]/scalar}"
            rot_transform_str = f"r {eul[0]} {eul[1]} {eul[2]}"
            svs_str = f"{geo_str} {pos_transform_str} {rot_transform_str}"
```